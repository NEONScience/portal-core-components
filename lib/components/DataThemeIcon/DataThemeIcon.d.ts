export default WrappedDataThemeIcon;
declare const WrappedDataThemeIcon: any;
