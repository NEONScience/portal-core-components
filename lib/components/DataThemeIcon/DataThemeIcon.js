"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _Avatar = _interopRequireDefault(require("@material-ui/core/Avatar"));
var _Theme = _interopRequireDefault(require("../Theme/Theme"));
var _atmosphere = _interopRequireDefault(require("./svg_optimized/atmosphere.svg"));
var _biogeochemistry = _interopRequireDefault(require("./svg_optimized/biogeochemistry.svg"));
var _ecohydrology = _interopRequireDefault(require("./svg_optimized/ecohydrology.svg"));
var _landcover = _interopRequireDefault(require("./svg_optimized/landcover.svg"));
var _organisms = _interopRequireDefault(require("./svg_optimized/organisms.svg"));
const _excluded = ["theme", "size", "avatar", "className"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.includes(n)) continue; t[n] = r[n]; } return t; }
const dataThemes = {
  atmosphere: {
    title: 'Atmosphere',
    aliases: ['atmos'],
    src: _atmosphere.default
  },
  biogeochemistry: {
    title: 'Biogeochemistry',
    aliases: ['biogeo'],
    src: _biogeochemistry.default
  },
  ecohydrology: {
    title: 'Ecohydrology',
    aliases: ['ecohydro'],
    src: _ecohydrology.default
  },
  landcover: {
    title: 'Land Cover & Processes',
    aliases: ['landuse', 'Land Cover and Processes', 'Land Use, Land Cover, and Land Processes'],
    src: _landcover.default
  },
  organisms: {
    title: 'Organisms, Populations, and Communities',
    aliases: [],
    src: _organisms.default
  }
};
const DataThemeIcon = props => {
  const {
      theme,
      size,
      avatar,
      className
    } = props,
    other = _objectWithoutPropertiesLoose(props, _excluded);
  const dataTheme = dataThemes[theme] || Object.values(dataThemes).find(entry => theme === entry.title || entry.aliases.includes(theme));
  if (!dataTheme) {
    return null;
  }
  const elementProps = {
    src: dataTheme.src,
    alt: dataTheme.title,
    title: dataTheme.title,
    className,
    'data-selenium': "data-theme-icon.".concat(theme)
  };
  if (avatar) {
    return /*#__PURE__*/_react.default.createElement(_Avatar.default, _extends({
      style: {
        width: _Theme.default.spacing(size),
        height: _Theme.default.spacing(size)
      }
    }, elementProps));
  }
  return /*#__PURE__*/_react.default.createElement("img", _extends({
    // eslint-disable-line jsx-a11y/alt-text
    width: _Theme.default.spacing(size),
    height: _Theme.default.spacing(size)
  }, elementProps, other));
};

// Data Themes do not have IDs. In order to support identifying a theme
// by its full title, abbreviated name, or other known aliases we build
// a full list of all supported aliases here to feed PropTypes.
const dataThemeAliases = [...Object.keys(dataThemes), ...Object.values(dataThemes).map(theme => theme.aliases.concat(theme.title)).flat()];
DataThemeIcon.propTypes = {
  theme: _propTypes.default.oneOf(dataThemeAliases).isRequired,
  size: _propTypes.default.number,
  avatar: _propTypes.default.bool,
  className: _propTypes.default.string
};
DataThemeIcon.defaultProps = {
  size: 5,
  avatar: false,
  className: null
};
const WrappedDataThemeIcon = _Theme.default.getWrappedComponent(DataThemeIcon);
var _default = exports.default = WrappedDataThemeIcon;