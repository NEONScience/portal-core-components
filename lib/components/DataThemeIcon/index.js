"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function () {
    return _DataThemeIcon.default;
  }
});
var _DataThemeIcon = _interopRequireDefault(require("./DataThemeIcon"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }