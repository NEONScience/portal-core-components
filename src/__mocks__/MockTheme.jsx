import React from 'react';

import CssBaseline from '@mui/material/CssBaseline';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';

import Theme from '../lib_components/components/Theme/Theme';

const MockTheme = ({ children }) => {
  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={Theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </StyledEngineProvider>
  );
}

export default MockTheme;
