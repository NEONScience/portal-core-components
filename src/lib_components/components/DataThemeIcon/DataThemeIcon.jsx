import React from 'react';
import PropTypes from 'prop-types';

import Avatar from '@material-ui/core/Avatar';

import Theme from '../Theme/Theme';

import AtmosphereSVG from './svg_optimized/atmosphere.svg';
import BiogeochemistrySVG from './svg_optimized/biogeochemistry.svg';
import EcohydrologySVG from './svg_optimized/ecohydrology.svg';
import LandCoverSVG from './svg_optimized/landcover.svg';
import OrganismsSVG from './svg_optimized/organisms.svg';
import ThemeDesc from '../../staticJSON/themeDesc.json'

const dataThemes = {
  atmosphere: {
    title: 'Atmosphere',
    aliases: ['atmos'],
    src: AtmosphereSVG,
    desc: ThemeDesc['Atmosphere'],
  },
  biogeochemistry: {
    title: 'Biogeochemistry',
    aliases: ['biogeo'],
    src: BiogeochemistrySVG,
    desc: ThemeDesc['Biogeochemistry'],
  },
  ecohydrology: {
    title: 'Ecohydrology',
    aliases: ['ecohydro'],
    src: EcohydrologySVG,
    desc: ThemeDesc['Ecohydrology'],
  },
  landcover: {
    title: 'Land Cover & Processes',
    aliases: ['landuse', 'Land Cover and Processes', 'Land Use, Land Cover, and Land Processes'],
    src: LandCoverSVG,
    desc: ThemeDesc['Land Cover & Processes'],
  },
  organisms: {
    title: 'Organisms, Populations, and Communities',
    aliases: [],
    src: OrganismsSVG,
    desc: ThemeDesc['Organisms, Populations, and Communities'],
  },
};

const DataThemeIcon = (props) => {
  const {
    theme,
    size,
    avatar,
    className,
    ...other
  } = props;
  const dataTheme = dataThemes[theme] || Object.values(dataThemes).find(
    (entry) => theme === entry.title || entry.aliases.includes(theme),
  );

  if (!dataTheme) { return null; }

  const elementProps = {
    src: dataTheme.src,
    alt: dataTheme.title,
    title: dataTheme.title,
    className,
    'data-selenium': `data-theme-icon.${theme}`,
    'data-desc': dataTheme.desc,
  };

  if (avatar) {
    return (
      <Avatar
        style={{ width: Theme.spacing(size), height: Theme.spacing(size) }}
        {...elementProps}
      />
    );
  }
  return (
    <img // eslint-disable-line jsx-a11y/alt-text
      width={Theme.spacing(size)}
      height={Theme.spacing(size)}
      {...elementProps}
      {...other}
    />
  );
};

// Data Themes do not have IDs. In order to support identifying a theme
// by its full title, abbreviated name, or other known aliases we build
// a full list of all supported aliases here to feed PropTypes.
const dataThemeAliases = [
  ...Object.keys(dataThemes),
  ...Object.values(dataThemes).map((theme) => theme.aliases.concat(theme.title)).flat(),
];

DataThemeIcon.propTypes = {
  theme: PropTypes.oneOf(dataThemeAliases).isRequired,
  size: PropTypes.number,
  avatar: PropTypes.bool,
  className: PropTypes.string,
};

DataThemeIcon.defaultProps = {
  size: 5,
  avatar: false,
  className: null,
};

const WrappedDataThemeIcon = Theme.getWrappedComponent(DataThemeIcon);

export default WrappedDataThemeIcon;
