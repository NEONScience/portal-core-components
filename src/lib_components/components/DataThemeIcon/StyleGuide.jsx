/* eslint react/jsx-one-expression-per-line: 0 */

import React from 'react';

import { makeStyles } from '@material-ui/core/styles';
import Divider from '@material-ui/core/Divider';
import Typography from '@material-ui/core/Typography';

import DocBlock from '../../../components/DocBlock';
import CodeBlock from '../../../components/CodeBlock';
import ExampleBlock from '../../../components/ExampleBlock';

import DataThemeIcon from './DataThemeIcon';
import Theme from '../Theme/Theme';

const useStyles = makeStyles(theme => ({
  divider: {
    margin: theme.spacing(3, 0),
  },
}));

export default function StyleGuide() {
  const classes = useStyles(Theme);
  return (
    <React.Fragment>

      <DocBlock>
        A set of five scalable icons representing the different NEON Data Themes.
      </DocBlock>
      <CodeBlock>
        {`
import DataThemeIcon from 'portal-core-components/lib/components/DataThemeIcon';
        `}
      </CodeBlock>

      <Typography variant="h5" component="h3" gutterBottom>Usage</Typography>

      <DocBlock>
        Use the <tt>theme</tt> prop to specify the theme.
      </DocBlock>
      <ExampleBlock>
        <DataThemeIcon theme="atmosphere" />
        <DataThemeIcon theme="biogeochemistry" />
        <DataThemeIcon theme="ecohydrology" />
        <DataThemeIcon theme="landcover" />
        <DataThemeIcon theme="organisms" />
      </ExampleBlock>
      <CodeBlock>
        {`
<DataThemeIcon theme="atmosphere" />
<DataThemeIcon theme="biogeochemistry" />
<DataThemeIcon theme="ecohydrology" />
<DataThemeIcon theme="landcover" />
<DataThemeIcon theme="organisms" />
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h6" component="h4" gutterBottom>Aliases</Typography>

      <DocBlock>
        Multiple aliases for the same theme are supported by the <tt>theme</tt> prop.
        The idea is, in lieu of theme IDs, to be able to feed in any string found
        in the wild to represent the theme and things should <i>just work</i>.
      </DocBlock>
      <ExampleBlock>
        <DataThemeIcon theme="landuse" />
        <DataThemeIcon theme="landcover" />
        <DataThemeIcon theme="Land Use, Land Cover, and Land Processes" />
      </ExampleBlock>
      <CodeBlock>
        {`
<DataThemeIcon theme="landuse" />
<DataThemeIcon theme="landcover" />
<DataThemeIcon theme="Land Use, Land Cover, and Land Processes" />
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h6" component="h4" gutterBottom>Size</Typography>

      <DocBlock>
        Use the <tt>size</tt> prop to scale the icon to any size relative to the
        standard <tt>theme.spacing</tt> unit (1 unit  = 8px). The default size is
        5 units, or 40 pixels.
      </DocBlock>
      <ExampleBlock>
        <DataThemeIcon theme="atmosphere" size={3} />
        <DataThemeIcon theme="atmosphere" />
        <DataThemeIcon theme="atmosphere" size={8} />
        <DataThemeIcon theme="atmosphere" size={20} />
      </ExampleBlock>
      <CodeBlock>
        {`
<DataThemeIcon theme="atmosphere" size={3} />
<DataThemeIcon theme="atmosphere" />
<DataThemeIcon theme="atmosphere" size={8} />
<DataThemeIcon theme="atmosphere" size={20} />
        `}
      </CodeBlock>

      <Divider className={classes.divider} />
      <Typography variant="h6" component="h4" gutterBottom>Avatar</Typography>

      <DocBlock>
        Use the <tt>avatar</tt> prop to toggle rendering as a Material UI Avatar
        instead of a standalone image. This renders the icon as a circle.
      </DocBlock>
      <ExampleBlock>
        <DataThemeIcon theme="atmosphere" size={3} avatar />
        <DataThemeIcon theme="biogeochemistry" size={6} avatar />
        <DataThemeIcon theme="ecohydrology" size={9} avatar />
        <DataThemeIcon theme="landcover" size={12} avatar />
        <DataThemeIcon theme="organisms" size={15} avatar />
      </ExampleBlock>
      <CodeBlock>
        {`
<DataThemeIcon theme="atmosphere" size={3} avatar />
<DataThemeIcon theme="biogeochemistry" size={6} avatar />
<DataThemeIcon theme="ecohydrology" size={9} avatar />
<DataThemeIcon theme="landcover" size={12} avatar />
<DataThemeIcon theme="organisms" size={15} avatar />
        `}
      </CodeBlock>
    </React.Fragment>
  );
}
