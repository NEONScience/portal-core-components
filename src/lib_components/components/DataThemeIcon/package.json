{
  "private": true,
  "name": "data-theme-icon",
  "main": "./DataThemeIcon.jsx",
  "module": "./DataThemeIcon.js"
}
