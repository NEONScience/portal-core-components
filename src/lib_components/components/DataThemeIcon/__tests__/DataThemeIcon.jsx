import React from 'react';
import renderer from 'react-test-renderer';

import MockTheme from '../../../../__mocks__/MockTheme';
import DataThemeIcon from '../DataThemeIcon';

describe('DataThemeIcon', () => {
  test('renders with only a theme prop', () => {
    const tree = renderer.create(
      <MockTheme>
        <DataThemeIcon theme="atmosphere" />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders with only a theme prop alias', () => {
    const tree = renderer.create(
      <MockTheme>
        <DataThemeIcon theme="biogeo" />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders with custom size', () => {
    const tree = renderer.create(
      <MockTheme>
        <DataThemeIcon theme="ecohydrology" size={4} />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders in avatar mode', () => {
    const tree = renderer.create(
      <MockTheme>
        <DataThemeIcon theme="landuse" avatar />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders with a custom className', () => {
    const tree = renderer.create(
      <MockTheme>
        <DataThemeIcon theme="organisms" className="foo" />
      </MockTheme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
