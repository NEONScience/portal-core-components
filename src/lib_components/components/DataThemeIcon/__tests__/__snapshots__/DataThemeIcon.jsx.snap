// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataThemeIcon renders in avatar mode 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circular css-avgvop-MuiAvatar-root"
  data-selenium="data-theme-icon.landuse"
  style={
    {
      "height": "40px",
      "width": "40px",
    }
  }
  title="Land Cover & Processes"
>
  <img
    alt="Land Cover & Processes"
    className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
    src="svg-file.svg"
  />
</div>
`;

exports[`DataThemeIcon renders with a custom className 1`] = `
<img
  alt="Organisms, Populations, and Communities"
  className="foo"
  data-selenium="data-theme-icon.organisms"
  height="40px"
  src="svg-file.svg"
  title="Organisms, Populations, and Communities"
  width="40px"
/>
`;

exports[`DataThemeIcon renders with custom size 1`] = `
<img
  alt="Ecohydrology"
  className={null}
  data-selenium="data-theme-icon.ecohydrology"
  height="32px"
  src="svg-file.svg"
  title="Ecohydrology"
  width="32px"
/>
`;

exports[`DataThemeIcon renders with only a theme prop 1`] = `
<img
  alt="Atmosphere"
  className={null}
  data-selenium="data-theme-icon.atmosphere"
  height="40px"
  src="svg-file.svg"
  title="Atmosphere"
  width="40px"
/>
`;

exports[`DataThemeIcon renders with only a theme prop alias 1`] = `
<img
  alt="Biogeochemistry"
  className={null}
  data-selenium="data-theme-icon.biogeo"
  height="40px"
  src="svg-file.svg"
  title="Biogeochemistry"
  width="40px"
/>
`;
